### DISCUSSION TITLE
guides/data-collection-and-annotation/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#13326] [Docs]
# guides/data-collection-and-annotation/

Data collection and annotation are vital steps in any computer vision project. Explore the tools, techniques, and best practices for collecting and annotating data.

https://docs.ultralytics.com/guides/data-collection-and-annotation/

**Comment by MoAbbasid** [comment]
Hello, Im looking for an open source annotation tool similar to labelme that exports directly to YOLOv8 format, something other than labelme?

**Reply by glenn-jocher** [reply]
@MoAbbasid hello! For an open-source annotation tool that exports directly to YOLOv8 format, you might want to try **CVAT**. It's quite versatile and supports exporting annotations directly into YOLO format among others. Another good option could be **Label Studio**, which offers flexibility in configuring export formats, including YOLO. Both tools provide robust features for efficient data labeling and are well-suited for various annotation tasks. Happy annotating! 😊

**Reply by MoAbbasid** [reply]
@glenn-jocher do they offer polygon annotations for instance segmentation?

**Reply by glenn-jocher** [reply]
Hello @MoAbbasid,

Yes, Ultralytics YOLOv8 does support polygon annotations for instance segmentation. This allows for more precise labeling of objects compared to traditional bounding boxes, which can be particularly useful for complex shapes and detailed segmentation tasks.

To create polygon annotations, you can use tools like **Roboflow Annotate** or **CVAT**. Both of these tools support polygon annotations and can export the data in a format compatible with YOLOv8.

Here's a brief overview of how you can use Roboflow for this purpose:

1. **Create a Project**: Start by creating a new project in Roboflow.
2. **Upload Images**: Upload your images to the project.
3. **Annotate Images**: Use the polygon tool to draw detailed outlines around objects. Roboflow also offers a SAM-based label assistant to speed up the annotation process.
4. **Export Annotations**: Once your annotations are complete, you can export the dataset in YOLOv8 format.

For more detailed steps on using Roboflow for data annotation and management, you can refer to the [Roboflow documentation](https://docs.ultralytics.com/integrations/roboflow/).

Here's a quick example of how to load and use a dataset with polygon annotations in YOLOv8:

```python
from ultralytics import YOLO

# Load a YOLOv8 model
model = YOLO('yolov8n-seg.pt')

# Train the model on your custom dataset
model.train(data='path/to/your/dataset.yaml', epochs=100, imgsz=640)

# Validate the model
metrics = model.val()

# Make predictions
results = model.predict(source='path/to/your/images')

# Visualize results
results.show()
```

Feel free to reach out if you have any more questions or need further assistance. Happy annotating! 😊

**Comment by ghost** [comment]
Hello, thanks for the info ! I would like to know how to perform active learning, specifically what are the metrics to choose the most informative images when the task is multi-instance multi-class object detection ?

**Reply by pderrenger** [reply]
@cawilchesp hello! Great question on active learning for multi-instance multi-class object detection. Active learning can significantly enhance your model by iteratively selecting the most informative samples for annotation. Here are some key metrics and strategies to consider:

### Metrics for Selecting Informative Images:
1. **Uncertainty Sampling**: Choose images where the model is least confident. This can be measured using:
   - **Entropy**: High entropy indicates uncertainty in predictions.
   - **Margin Sampling**: Select samples where the difference between the top two class probabilities is smallest.
   - **Least Confidence**: Select samples with the lowest maximum class probability.

2. **Diversity Sampling**: Ensure a diverse set of samples to avoid redundancy.
   - **Core-Set Selection**: Use clustering techniques to select a diverse subset of data points.
   - **Representative Sampling**: Choose samples that represent the overall data distribution.

3. **Error-Based Sampling**: Focus on samples where the model previously made errors.
   - **Misclassified Samples**: Select images where the model's predictions were incorrect.
   - **High Loss Samples**: Choose samples with high loss values during training.

### Implementing Active Learning:
Here's a basic example using uncertainty sampling with entropy:

```python
import numpy as np
from ultralytics import YOLO

# Load your model
model = YOLO('yolov8n.pt')

# Function to calculate entropy
def calculate_entropy(probs):
    return -np.sum(probs * np.log(probs), axis=1)

# Predict on a batch of images
results = model(['image1.jpg', 'image2.jpg', 'image3.jpg'])

# Calculate entropy for each image
entropies = [calculate_entropy(result.probs) for result in results]

# Select the image with the highest entropy
most_informative_image = results[np.argmax(entropies)]
```

### Additional Resources:
For more detailed strategies and examples, you can refer to our [Data Collection and Annotation Strategies](https://docs.ultralytics.com/guides/data-collection-and-annotation/).

Feel free to experiment with these methods and see which works best for your specific use case. If you have any further questions or need more examples, don't hesitate to ask! 😊

**Comment by robert-hardman** [comment]
Hello
In object recognition with YOLO11
If we label images by drawing polygons:
When training, does YOLO11 convert polygon coordinates to boxes?
How does Inference handle polygon coordinates?

**Reply by glenn-jocher** [reply]
@robert-hardman yOLO11 does not inherently convert polygon annotations to bounding boxes during training. Polygons are used directly for tasks like instance segmentation, while bounding boxes are utilized for object detection. During inference, polygon annotations are retained for segmentation tasks, ensuring accurate object shape representation. For more on annotation types, refer to [Data Annotation Guide](https://docs.ultralytics.com/guides/data-collection-and-annotation/).

**Comment by Team-Q-Botix** [comment]
I’m working on Robocon 2026 (Kung Fu Quest) and using YOLO-seg for detecting KFS symbols — ancient Chinese oracle bone characters.
There are 15 true symbols, 15 fake ones (random patterns mimicking the real ones), plus 1 “other” class → total 31 classes.

Do you think instance segmentation (YOLOv8-seg / YOLOv11-seg) is the right approach for these detailed 2D symbols, or would box detection + contour refinement be better?
Also, any advice on which model to start with and how large a dataset (per class) you’d recommend for good mask performance on Jetson Orin?

**Comment by Sudharsanam-1** [comment]
hello ! For detecting 31 Chinese symbols (16 true and 15 fake), how many classes should be used in the model?
 Is it more efficient to detect  the true symbols using only  true class , or should we also include the fake symbols as separate classes?

**Comment by leticiazorzirama** [comment]
Hello,

Has anyone worked with oriented object detection using a multisource dataset that combines satellite and UAV imagery?

I am currently developing a scientific project focused on oriented detection of fishing vessels. As part of this work, I conducted two systematic reviews: one covering existing oriented detection models and another covering publicly available datasets. So far, I have identified multisource datasets that combine imagery from multiple satellite platforms; however, I have not found datasets that explicitly integrate both satellite and UAV images.

I understand the importance of source diversity to mitigate dataset bias and improve model generalization. That said, I would appreciate insights from the community on whether combining UAV and satellite imagery might introduce excessive domain variability (e.g., differences in spatial resolution, viewing angles, and sensor characteristics), potentially hindering model training and convergence.

Any experiences, references, or best practices related to this type of multisource setup would be highly appreciated.

**Reply by glenn-jocher** [reply]
Great question—mixing UAV + satellite can increase domain shift (GSD/resolution, look-angle, sensor PSF), but it’s usually manageable if you control the dataset and evaluation: keep one unified, very explicit labeling policy (same object extent/rotation rules) and balanced sampling as described in our [Data Collection and Annotation guide](https://docs.ultralytics.com/guides/data-collection-and-annotation/), then train with per-source “slice” validation (sat-only/UAV-only) and consider staged training (pretrain on the larger domain, then fine-tune on the smaller) rather than a single fully-mixed run; if you’re using Ultralytics YOLO for oriented detection, the [OBB task docs](https://docs.ultralytics.com/tasks/obb/) cover the expected label format and training flow—feel free to share your planned split strategy and a couple example tiles (UAV + satellite) here and the team/community can sanity-check the setup.

**Reply by leticiazorzirama** [reply]
Hi Glenn, 

To be more specific,

I am using a pretrained yolo26 m obb to perform detection of a single class: ships of a particular type (fishing boats).

I have a dataset with satellite optical images collected from google earth pro counting 331 pairs of images/labels.
The total number of instances is 5974.
Images size are fixed at 4823 height x 8192 width.
The bounding boxes have 0.075 height x 0.04 width (according to yolo train results).
Resolution varies from very-low to very-high.
The images were captured with an altitude between 200-400 meters.

I also have 21 very-high resolution images collected from UAV. 
Images size are fixed at 3072 height x 4096 width.
The images were captured with an altitude of 120 meters.

I would like to test if incorporating UAV images in training data would improved the models hability to detect the objects.

From what you proposed, please confirm if I understood correctly: 

- Should the train set be mixed (sat+UAV) and the validation set be separate (sat-only/UAV-only)?
- If so, is this separation configured the staged training you are refering? That is, pretrain with sat+UAV as the train set and sat-only as validation subset, and further fine-tune with the same train subset (sat+UAV) but now using the uav-only as the validation set?

Thanks

**Reply by leticiazorzirama** [reply]
OBS.: sorry, sometimes I wrote 'train subset', and 'validation subset' but all of them was supposed to be written as 'train set', and 'validation set'.

**Reply by glenn-jocher** [reply]
Your first interpretation is correct: compare a satellite-only baseline against mixed satellite+UAV training while keeping fixed, disjoint satellite-only and UAV-only evaluation sets. This is not staged training—changing only the validation set does not fine-tune the model. Staged training would first train on satellite data, then continue from that checkpoint using only the UAV training split with a lower learning rate, evaluating both domains afterward. With only 21 UAV images, split by original scene before tiling to prevent leakage, and consider grouped k-fold validation because a single UAV split will be unstable.

**Reply by leticiazorzirama** [reply]
Understood, thanks.
